walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,189 @@
+            }
+        }
+    }
+
+    
+    //ignores people not attending
+    if(attendeeSchedule.size()==0){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+        
+
+
+        List<TimeRange> busySchedule = new ArrayList<TimeRange>();
+            Iterator<TimeRange> myIt = attendeeSchedule.iterator(); 
+            while (myIt.hasNext()){
+                busySchedule.add(myIt.next());
+                
+            }
+
+        //sort times that don't work
+        Collections.sort(busySchedule, TimeRange.ORDER_BY_END);
+
+        //time nodes that are contained in others
+        trimContainedNodes(busySchedule);
+
+        
+
+        //test
+        for(int k=0 ; k < busySchedule.size(); k++){
+
+            TimeRange currentTimeRange = busySchedule.get(k);
+            int start = currentTimeRange.start();
+            int end = currentTimeRange.end();
+            int duration = currentTimeRange.duration();
+
+            if(k == 0 && busySchedule.size()!=0){
+        
+                TimeRange possibleTime = TimeRange.fromStartEnd(dayStart,start,false);
+                
+                        if(possibleTime.duration() >= meetingDuration){
+                        possibleTimes.add(possibleTime);
+                        
+                        
+                    }
+                
+            }
+
+            if(k == busySchedule.size()-1 && busySchedule.size()!=0){
+                
+                TimeRange possibleTime = TimeRange.fromStartEnd(end,dayEnd,true);
+
+                if(k!=0){
+                    int possibleStartTime = Math.max(end,possibleTime.end());
+                    TimeRange anotherPossibleTime = TimeRange.fromStartEnd(end,dayEnd,true);
+
+                    if(anotherPossibleTime.overlaps(possibleTime)){
+                        if(anotherPossibleTime.duration() >= meetingDuration){
+                            possibleTimes.add(anotherPossibleTime);
+                        
+                        }
+
+                    }
+                    
+                }else{
+                    if(possibleTime.duration() >= meetingDuration){
+                        possibleTimes.add(possibleTime); 
+                        
+                    }
+                }     
+            }
+
+            if(k+1<busySchedule.size()){
+                
+                TimeRange possibleTime = TimeRange.fromStartEnd(end,busySchedule.get(k+1).start(),false);
+                
+                    if(possibleTime.duration() >= meetingDuration){
+                    possibleTimes.add(possibleTime);
+
+                    
+                    
+                }
+            }
+
+        }
+        
+
+
+    
+     return possibleTimes;
   }
+
+    //check if person is attendee of event
+   public boolean isAnAttendee(String attendee, Event event){
+       if (event.getAttendees().contains(attendee)){
```

Comment:
You could shorten this function to one line:

`return event.getAttendees().contains(attendee)`

And at that point, you might consider just using that code directly instead of putting it in this helper function.